python/python/tests/test_config.py

```diff
@@ -52,3 +57,195 @@ def test_periodic_checks_interval_to_protobuf():
+            route=AllNodes(),
+        )
+        await client_for_config_set.config_resetstat() == OK
+
+        # 12182 is the slot of "foo"
+        await client_for_config_set.custom_command(
+            ["CONFIG", "SET", "availability-zone", az],
+            route=SlotIdRoute(SlotType.REPLICA, 12182),
+        )
+
+        await client_for_config_set.close()
+
+        client_for_testing_az = await create_client(
+            request,
+            cluster_mode,
+            protocol=protocol,
+            read_from=ReadFrom.AZ_AFFINITY,
+            timeout=2000,
+            client_az=az,
+        )
+        for _ in range(GET_CALLS):
+            await client_for_testing_az.get("foo")
+
+        info_result = await client_for_testing_az.custom_command(
+            ["INFO", "ALL"], route=AllNodes()
+        )
+
+        # Check that only the replica with az has all the GET calls
+        matching_entries_count = sum(
+            1
+            for value in info_result.values()
+            if get_cmdstat in value.decode() and az in value.decode()
+        )
+        assert matching_entries_count == 1
+
+        # Check that the other replicas have no availability zone set
+        changed_az_count = sum(
+            1
+            for node in info_result.values()
+            if f"availability_zone:{az}" in node.decode()
+        )
+        assert changed_az_count == 1
+
+        await client_for_testing_az.close()
+
+    @pytest.mark.skip_if_version_below("8.0.0")
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_az_affinity_non_existing_az(
+        self, request, cluster_mode: bool, protocol: ProtocolVersion
+    ):
+        GET_CALLS = 4
+        get_cmdstat = f"cmdstat_get:calls={GET_CALLS}"
+
+        client_for_testing_az = await create_client(
+            request,
+            cluster_mode,
+            protocol=protocol,
+            read_from=ReadFrom.AZ_AFFINITY,
+            timeout=2000,
+            client_az="non-existing-az",
+        )
+        await client_for_testing_az.config_resetstat() == OK
+
+        for _ in range(GET_CALLS):
+            await client_for_testing_az.get("foo")
+
+        info_result = await client_for_testing_az.custom_command(
+            ["INFO", "ALL"], route=AllNodes()
+        )
+
+        # Check that a replica has all the GET calls
+        matching_entries_count = sum(
+            1 for value in info_result.values() if get_cmdstat in value.decode()
+        )
+        assert matching_entries_count == 1
+
+        await client_for_testing_az.close()
+
+    @pytest.mark.skip_if_version_below("8.0.0")
+    @pytest.mark.parametrize("cluster_mode", [True])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_az_affinity_requires_client_az(
+        self, request, cluster_mode: bool, protocol: ProtocolVersion
```

Comment:
question: where does the request value get initialized?

Reply:
I think it's initialized by pytest itself. It is a built-in pytest fixture that is automatically injected into the test functions, and you do not need to manually initialize it.

Reply:
see: https://docs.pytest.org/en/stable/how-to/fixtures.html#request-context